### PR TITLE
Support for renaming subdirectories

### PR DESCRIPTION
**Pull request by djalova**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by djalova** [comment]
For issue #113 
@gilv Can you review this please?

File: src/main/java/com/ibm/stocator/fs/ObjectStoreFileSystem.java [hunk 1]
```diff
@@ -218,10 +218,24 @@ public boolean rename(Path src, Path dst) throws IOException {
       return true;
     }
     LOG.debug("Checking if source exists {}", src);
+
     if (exists(src)) {
-      LOG.debug("Source {} exists", src);
+      boolean result = storageClient.rename(hostNameScheme, src.toString(), dst.toString());
+
+      // Check if path was a directory
+      FileStatus[] files = storageClient.list(hostNameScheme, src, true, true);
```
Comment (gilv) [inline src/main/java/com/ibm/stocator/fs/ObjectStoreFileSystem.java]: @djalova Can you please call storageClient getFileStatus and see if it's directory or not? This way we will not duplicate code to figure out if the given path is dir or file. thanks

File: src/main/java/com/ibm/stocator/fs/ObjectStoreFileSystem.java [hunk 2]
```diff
@@ -218,10 +218,24 @@ public boolean rename(Path src, Path dst) throws IOException {
       return true;
     }
     LOG.debug("Checking if source exists {}", src);
+
     if (exists(src)) {
-      LOG.debug("Source {} exists", src);
+      boolean result = storageClient.rename(hostNameScheme, src.toString(), dst.toString());
+
```
Comment (gilv) [inline src/main/java/com/ibm/stocator/fs/ObjectStoreFileSystem.java]: @djalova this should be called if src is not exists, right?

Comment (djalova) [inline src/main/java/com/ibm/stocator/fs/ObjectStoreFileSystem.java]: @gilv Can you explain what you mean? If src does not exist, then we return false without doing anything.

File: src/main/java/com/ibm/stocator/fs/ObjectStoreFileSystem.java (hunk 1 above)
Comment (djalova) [inline src/main/java/com/ibm/stocator/fs/ObjectStoreFileSystem.java]: @gilv Updated the PR

**Comment by gilv** [comment]
We don't need rename on directory level. Rename on the object level supports only _spark_metadata by now

**Pull request closed without merging**